teamcity: failed tests on master: acceptance/TestDockerCLI, acceptance/TestDockerCLI/test_sql_mem_monitor.tcl

The following tests appear to have failed:

[#238928](https://teamcity.cockroachdb.com/viewLog.html?buildId=238928):

```
--- FAIL: acceptance/TestDockerCLI (147.520s)
test_log_scope.go:80: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestDockerCLI350710226
test_log_scope.go:63: use -show-logs to present logs inline
--- FAIL: acceptance/TestDockerCLI/test_sql_mem_monitor.tcl (35.800s)
cli_test.go:73: non-zero exit code: 1
------- Stdout: -------
*
* INFO: command left-over files in /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestDockerCLI/test_sql_mem_monitor.tcl-c264fb2b
*

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/acceptance/cli_test.go
+++ b/pkg/acceptance/cli_test.go
@@ -33,7 +33,8 @@ const containerPath = "/go/src/github.com/cockroachdb/cockroach/cli/interactive_
 var cmdBase = []string{
 	"/usr/bin/env",
 	"COCKROACH_SKIP_UPDATE_CHECK=1",
-	"/usr/bin/expect",
+	"/bin/bash",
+	"-c",
 }
 
 func TestDockerCLI(t *testing.T) {
@@ -63,12 +64,22 @@ func TestDockerCLI(t *testing.T) {
 		testPath := filepath.Join(containerPath, testFile)
 		t.Run(testFile, func(t *testing.T) {
 			log.Infof(ctx, "-- starting tests from: %s", testFile)
-			cmd := cmdBase
+
+			// We run the expect command using `bash -c "(expect ...)"`.
+			//
+			// We cannot run `expect` directly, nor `bash -c "expect ..."`,
+			// because both cause Expect to become the PID 1 process inside
+			// the container. On Unix, orphan processes need to be wait()ed
+			// upon by the PID 1 process when they terminate, lest they
+			// remain forever in the zombie state. Unfortunately, Expect
+			// does not contain code to do this. Bash does.
+			cmd := "(expect"
 			if verbose {
-				cmd = append(cmd, "-d")
+				cmd = cmd + " -d"
 			}
-			cmd = append(cmd, "-f", testPath, cluster.CockroachBinaryInContainer)
-			containerConfig.Cmd = cmd
+			cmd = cmd + " -f " + testPath + " " + cluster.CockroachBinaryInContainer + ")"
+			containerConfig.Cmd = append(cmdBase, cmd)
+
 			if err := testDockerOneShot(ctx, t, "cli_test", containerConfig); err != nil {
 				t.Error(err)
 			}

--- a/pkg/cli/interactive_tests/common.tcl
+++ b/pkg/cli/interactive_tests/common.tcl
@@ -85,12 +85,15 @@ proc start_server {argv} {
 }
 proc stop_server {argv} {
     report "BEGIN STOP SERVER"
+    # Trigger a normal shutdown.
     system "$argv quit"
+    # If after 5 seconds the server hasn't shut down, trigger an error.
+    system "for i in `seq 1 5`; do kill -CONT `cat server_pid` 2>/dev/null || exit 0; echo still waiting; sleep 1; done; echo 'server still running?'; exit 1"
     report "END STOP SERVER"
 }
 
 proc force_stop_server {argv} {
     report "BEGIN FORCE STOP SERVER"
-    system "set -x; $argv quit & sleep 1; if kill -CONT `cat server_pid`; then kill -TERM `cat server pid`; sleep 1; if kill -CONT `cat server_pid`; then kill -KILL `cat server_pid`; fi; fi"
+    system "$argv quit & sleep 1; if kill -CONT `cat server_pid` 2>/dev/null; then kill -TERM `cat server pid`; sleep 1; if kill -CONT `cat server_pid` 2>/dev/null; then kill -KILL `cat server_pid`; fi; fi"
     report "END FORCE STOP SERVER"
 }

